source/speech.py

```diff
@@ -1939,3 +2008,489 @@ def __repr__(self):
 		if self.text:
 			out += ", text=%r" % self.text
 		return out + ")"
+
+class BaseCallbackCommand(SpeechCommand):
+	"""Base class for commands which cause a function to be called when speech reaches them.
+	This class should not be instantiated directly.
+	It is designed to be subclassed to provide specific functionality;
+	e.g. L{BeepCommand}.
+	To supply a generic function to run, use L{CallbackCommand}.
+	This command is never passed to synth drivers.
+	"""
+
+	def run(self):
+		"""Code to run when speech reaches this command.
+		"""
+
+class CallbackCommand(BaseCallbackCommand):
+	"""Call a function when speech reaches this point.
+	"""
+
+	def __init__(self, callback):
+		self.run = callback
+
+class BeepCommand(BaseCallbackCommand):
+	"""Produce a beep.
+	"""
+
+	def __init__(self, hz, length, left=50, right=50):
+		self.hz = hz
+		self.length = length
+		self.left = left
+		self.right = right
+
+	def run(self):
+		import tones
+		tones.beep(self.hz, self.length, left=self.left, right=self.right)
+
+	def __repr__(self):
+		return "BeepCommand({hz}, {length}, left={left}, right={right})".format(
+			hz=self.hz, length=self.length, left=self.left, right=self.right)
+
+class WaveFileCommand(BaseCallbackCommand):
+	"""Play a wave file.
+	"""
+
+	def __init__(self, fileName):
+		self.fileName = fileName
+
+	def run(self):
+		import nvwave
+		nvwave.playWaveFile(self.fileName, async=False)

```

Comment:
According to @jcsteh in https://github.com/nvaccess/nvda/pull/7599#issuecomment-389812530, async should be True